InputValidation/ValidationMethods/Validation.FirstCharacter.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using TotalImage.Properties;
+using TotalImage.Validation;
+
+/// <summary>
+/// Validates the first character of a DOS 8.3 filename (in the case it has been overwritten with 0xE5) (2020/08/01)
+/// </summary>
+namespace TotalImage.Validation.ValidationMethods
+{
+    public class FirstCharacterValidation : IValidation
+    {
+        public object Validate(object Data) 
+        {
+            // Return if not char
+            Data = ValidationCore(Data);
+
+            if (Data == null)
+            {
+                OnFail();
+            }
+            else
+            {
+                return Data;
+            }
+
+            //blah
+            return null; 
+
+        }
+
+        private object ValidationCore(object Data)
+        {
+            if (!VerifyDataType(Data))
+            {
+                return null;
+            }
+            else
+            {
+                // Convert to character. It must be a char by this point. 
+                char D = (char)Data;
+
+                // Control characters not allowed.
+                if (D <= 0x1F) return null;
+
+                // Other invalid characters. (what to do about space?)
+                if (D == '#'
+                    || D == '*'
+                    || D == '+'
+                    || D == ','
+                    || D == '/'
+                    || D == ':'
+                    || D == ';'
+                    || D == '<'
+                    || D == '='
+                    || D == '>'
+                    || D == '?'
+                    || D == '\\' // requires escaping 
+                    || D == '['
+                    || D == ']'
+                    || D == '|') return null;
+
+                // Lowercase to uppercase. 
+                if (D >= 0x61 || D <= 0x7A)
+                {
+                    // why isn't (D -= 0x20) valid? 
+
+                    // Alternatively you could use a string and ToLowerInvariant to do this.
+                    byte _ = (byte)D;
+                    _ -= 0x20;
+                    D = (char)_;
```

Comment:
In this case,
```c#
D -= (char)0x20
```
should work, but I think `char.ToLowerInvariant(D)` would be the preferred option here. This is because core methods like this in .NET often have special lower-level optimisations associated with them and as such, perform better than doing the same thing more manually. It's also clearer about the intent even without comments.

While still technically legal, using an underscore as a variable name is not best practice any more because of the introduction of [discards](https://docs.microsoft.com/en-us/dotnet/csharp/discards) which use underscores to signify a return value that can be safely ignored.